be/src/exec/vectorized/olap_scan_node.cpp

```diff
@@ -642,8 +644,27 @@ void OlapScanNode::_close_pending_scanners() {
 }
 
 pipeline::OpFactories OlapScanNode::decompose_to_pipeline(pipeline::PipelineBuilderContext* context) {
-    auto factory = std::make_shared<pipeline::OlapScanOperatorFactory>(context->next_operator_id(), this);
-    return pipeline::decompose_scan_node_to_pipeline(factory, this, context);
+    auto scan_ctx = std::make_shared<pipeline::OlapScanContext>(this);
+    auto&& rc_rf_probe_collector = std::make_shared<RcRfProbeCollector>(2, std::move(this->runtime_filter_collector()));
+
+    // scan_prepare_op.
+    auto scan_prepare_op =
```

Comment:
Is there will be performance issue if add additional two operators?

Reply:
I test on SSB single dataset, and there isn't performance regression.
The only one additional work is schedule of this two operators.